sdcm/utils/url_validate.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+import requests
+
+SCYLLA_URL_REGEX = re.compile('.*(?P<url>http.*scylladb.com.*).*')
+URL_REGEX_AND_FORMAT_LIST = [
+    (re.compile(r'baseurl=(?P<url>http.*scylladb.com.*)'), '{url}repodata/repomd.xml'),
+    (re.compile(r'deb\s+\[arch=(?P<arch>.*?)\]\s(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$'),
+     "{url}/dists/{version_code_name}/{component}/binary-{arch}/Packages"),
+    (re.compile(r'deb\s+(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$'),
+     "{url}/dists/{version_code_name}/{component}/binary-amd64/Packages.gz"),
+]
+URL_ENV_VARIABLES = {
+    '$basearch': 'x86_64',
+    '$releasever': '7',
+}
+
+
+def verify_url_is_valid(url):
+    """
+    Check the URL of `scylla.list` is valid and verify it's a file.
+    After that check, the file contains a valid scheme and package URL.
+    """
+    package_urls, error_msg = [], "Not find any Scylla URL version"
+    for url_version in requests.get(url, stream=True).text.splitlines():
+        if not SCYLLA_URL_REGEX.match(url_version):
+            continue
+        for (url_regex, url_format) in URL_REGEX_AND_FORMAT_LIST:
+            packages_url = url_regex.match(url_version)
+            if packages_url:
+                package_url = url_format.format(**packages_url.groupdict())
+                for env_name, env_value in URL_ENV_VARIABLES.items():
+                    package_url = package_url.replace(env_name, env_value)
+                if requests.get(package_url).status_code == 200:
+                    package_urls.append(package_url)
+                    break
+        else:
+            raise ValueError(error_msg)
+    if not package_urls:
+        raise ValueError(error_msg)
+    return package_urls[0]
```

Comment:
what is the purpose of this return?

Reply:
I tried to supported in old logic, Ill change the logic of  `get_branch_version_from_list` and `get_branch_version_from_repo`